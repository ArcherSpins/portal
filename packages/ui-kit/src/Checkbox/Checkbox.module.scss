@import "../variables.scss";

.checkbox {
  font-size: 0.875rem;
  display: flex;

  input {
    display: none;
  }

  b {
    margin-left: 0.625rem;
  }

  i {
    position: absolute;
  }

  label {
    display: inline-flex;
    align-items: center;
    cursor: pointer;
  }

  &__icon {
    color: $main-color;
    border: 1px solid $border-color;
    border-radius: 6px;
    box-sizing: border-box;
    display: inline-flex;
    justify-content: center;
    align-items: center;
    font-size: 1rem;
    width: 1.5rem;
    height: 1.5rem;
    position: relative;
  }
}
