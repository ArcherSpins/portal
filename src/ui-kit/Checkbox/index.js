// @flow

import React from 'react';
import styles from './Checkbox.module.scss';

type CheckboxProps = {
  id: string,
  checked: boolean,
  label?: string,
  onChange: (e: SyntheticEvent<HTMLInputElement>) => void,
};

const Checkbox = ({
  label,
  id,
  checked,
  onChange,
}: CheckboxProps) => (
  <div className={styles.checkbox}>
    <label htmlFor={id}>
      <input
        id={id}
        onChange={onChange}
        type="checkbox"
        checked={checked}
      />
      <span className={styles.checkbox__icon}>
        {checked && <i className="icon-check" />}
      </span>
      {label && label.length > 0 && (<b>{label}</b>)}
    </label>
  </div>
);

Checkbox.defaultProps = {
  label: '',
};

export default Checkbox;
