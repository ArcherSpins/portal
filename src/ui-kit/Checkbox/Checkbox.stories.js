// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import Checkbox from 'ui-kit/Checkbox';

storiesOf('Checkbox', module)
  .addDecorator(withKnobs)
  // TODO: $FlowFixMe
  .addWithChapters('Default', {
    chapters: [
      {
        sections: [
          {
            title: 'Idle',
            sectionFn: () => {
              const id = 'checkbox_1';

              return (
                <Checkbox
                  id={id}
                  checked={boolean('Checked', false)}
                  onChange={action(`On checkbox #${id} click`)}
                />
              );
            },
          },
          {
            title: 'With label',
            sectionFn: () => {
              const id = 'checkbox_2';

              return (
                <Checkbox
                  id="checkbox 2"
                  checked={boolean('Checked', false)}
                  label="Inactive tasks"
                  onChange={action(`On checkbox #${id} click`)}
                />
              );
            },
          },
        ],
      },
    ],
  });
