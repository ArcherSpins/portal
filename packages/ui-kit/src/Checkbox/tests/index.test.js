// @flow

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import React from 'react';
import Checkbox from '..';

const defaultProps = {
  id: 'checkbox',
  checked: false,
  onChange: () => {},
};

describe('The Checkbox component', () => {
  it('renders correctly', () => {
    const component = (
      <Checkbox
        {...defaultProps}
      />
    );
    const wrapper = shallow(component);
    expect(wrapper.hasClass('checkbox')).toBeTruthy();
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have default props', () => {
    const wrapper = shallow(<Checkbox {...defaultProps} />);
    expect(wrapper.find('label b').length).toEqual(0);
  });

  it('should have label', () => {
    const label = 'Inactive tasks';
    const wrapper = shallow(
      <Checkbox
        {...defaultProps}
        label={label}
      />,
    );
    expect(wrapper.find('label b').prop('children')).toEqual(label);
  });

  it('should change state', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <Checkbox
        {...defaultProps}
        onChange={onChange}
      />,
    );
    const input = wrapper.find('label input').first();
    expect(input.prop('onChange')).toEqual(onChange);
    input.simulate('change', { target: { checked: true } });
    expect(onChange).toHaveBeenCalled();
  });
});
