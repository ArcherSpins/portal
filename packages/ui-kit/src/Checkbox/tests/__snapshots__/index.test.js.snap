// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Checkbox component renders correctly 1`] = `
<div
  className="checkbox"
>
  <label
    htmlFor="checkbox"
  >
    <input
      checked={false}
      id="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="checkbox__icon"
    />
    
  </label>
</div>
`;
